client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/index.tsx

```diff
@@ -127,10 +128,19 @@ export default function PlansStepAdaptor( props: StepProps ) {
 	const isWordCampPromo = new URLSearchParams( location.search ).has( 'utm_source', 'wordcamp' );
 	const plansIntent = getPlansIntent( props.flow, isWordCampPromo );
 
-	const hidePlanProps =
-		createWithBigSky && isGoalFirstExperiment
-			? getHidePlanPropsBasedOnCreateWithBigSky()
-			: getHidePlanPropsBasedOnThemeType( selectedThemeType || '' );
+	let hidePlanProps;
+	if ( createWithBigSky && isGoalFirstExperiment ) {
+		hidePlanProps = getHidePlanPropsBasedOnCreateWithBigSky();
```

Comment:
Can we extend this function to do it instead? It seems related.

Reply:
Honestly I struggle to make it reusable without adding more complexity than it already has. The function would need 3 parameters to return 3 lines and I don't know if it's worth it.
Here is how it shakes out:

```
	let hidePlanProps;
	if ( createWithBigSky && isGoalFirstExperiment ) {
		hidePlanProps = {
			hideFreePlan: true,
			hideEcommercePlan: true,
			hideEnterprisePlan: true,
		};
	} else if ( props.flow === AI_SITE_BUILDER_FLOW ) {
		hidePlanProps = {
			hideFreePlan: true,
			hidePersonalPlan: true,
			hideEcommercePlan: true,
			hideEnterprisePlan: true,
		};
	} else {
		hidePlanProps = getHidePlanPropsBasedOnThemeType( selectedThemeType || '' );
	}
	```